web/client/api/ArcGIS.js

```diff
@@ -87,12 +88,21 @@ export const searchAndPaginate = (records, params) => {
     };
 };
 const getData = (url, params = {}) => {
+    let headers = {};
+    const protectedId = params?.info?.options?.service?.protectedId;
+    const storedProtectedService = getCredentials(protectedId);
+    if (storedProtectedService) {
+        headers = {
+            "Authorization": `Basic ${btoa(storedProtectedService.username + ":" + storedProtectedService.password)}`
+        };
+    }
```

Comment:
This same operation is repeated multiple times in this PR, probably we could create a function in SecurityUtils that given a service it returns headers.

Reply:
sure